library/src/scala/internal/quoted/Matcher.scala

```diff
@@ -271,13 +271,16 @@ private[quoted] object Matcher {
           case (scrutinee, Typed(expr2, _)) =>
             scrutinee =?= expr2
 
-
           /* Match selection */
           case (Select(qual1, _), Select(qual2, _)) if scrutinee.symbol == pattern.symbol =>
             qual1 =?= qual2
+          case (ref: Ident, Select(qual2, _)) if scrutinee.symbol == pattern.symbol =>
+            Ref.desugarIdent(ref) match
+              case Select(qual1, _) => qual1 =?= qual2
+              case _ => matched
```

Comment:
Or do a type test instead?

Reply:
No, I need to do the recursion because `qual2` may contain a term hole that needs to be matched.